casanovo/data/ms_io.py

```diff
@@ -2,17 +2,99 @@
 
 import collections
 import csv
+import logging
 import operator
 import os
 import re
+from datetime import datetime
 from pathlib import Path
-from typing import List
+from socket import gethostname
+from sys import argv
+from time import time
+from typing import List, Optional, Dict
 
 import natsort
+import numpy as np
+import torch
+from pandas import DataFrame
 
 from .. import __version__
 from ..config import Config
 
+SCORE_BINS = [0.0, 0.5, 0.9, 0.95, 0.99]
+
+logger = logging.getLogger("casanovo")
+
+
+def get_score_bins(
+    results_table: DataFrame, score_bins: List[float]
+) -> Dict[float, int]:
+    """
+    From a list of confidence scores, return a dictionary mapping each confidence score
+    to the number of spectra with a confidence greater than or equal to it.
+
+    Parameters
+    ----------
+        results_table: DataFrame
+            Parsed spectrum match table
+        score_bins: List[float]
+            Confidence scores to map
+
+    Returns
+    -------
+        score_bin_dict: Dict[float, int]
+            Dictionary mapping each confidence score to the number of spectra with a confidence
+            greater than or equal to it.
+    """
+    se_scores = results_table["score"].to_numpy()
```

Comment:
**suggestion:** [`Series.array`](https://pandas.pydata.org/docs/reference/api/pandas.Series.array.html) is recommended. But I don't think this conversion is even necessary, it can operate on the Series directly?